File: 1615-maximal-network-rank/1615-maximal-network-rank.cpp
class Solution {
public:
    int maximalNetworkRank(int n, vector<vector<int>>& roads) {
        vector<int> adjList[n];
        vector<vector<int>> Map(n, vector<int>(n));
        for(vector<int> edge: roads) {
            adjList[edge[0]].push_back(edge[1]);
            adjList[edge[1]].push_back(edge[0]);
            Map[edge[0]][edge[1]] = 1;
            Map[edge[1]][edge[0]] = 1;
        }
        int ans = 0;
        for(int i = 0; i < n; i++) {
            for(int j = i + 1; j < n; j++) {
                ans = max(ans, (int)adjList[i].size() + (int)adjList[j].size() - Map[i][j]);
            }
        }
        return ans;
    }
};

File: 1615-maximal-network-rank/1615. Maximal Network Rank.go
func max(a, b int) int {
  if a > b {return a}
  return b
}

func maximalNetworkRank(n int, roads [][]int) int {
      degree := make([]int, n)
      Map := make([][]int, n)
      for i := 0; i < n; i++ {
          Map[i] = make([]int, n)
      }
      for _, edge := range roads {
          degree[edge[0]]++
          degree[edge[1]]++
          Map[edge[0]][edge[1]] = 1
          Map[edge[1]][edge[0]] = 1
      }
      ans := 0
      for i := 0; i < n; i++ {
          for j := i + 1; j < n; j++ {
              ans = max(ans, degree[i] + degree[j] - Map[i][j])
          }
      }
      return ans
}
